api/queue/work/model_service_deployment.go

```diff
@@ -58,17 +56,8 @@ func (depl *ModelServiceDeployment) Deploy(job *queue.Job) error {
 		return err
 	}
 
-	endpointArg := jobArgs.Endpoint
-	endpoint, err := depl.Storage.Get(endpointArg.ID)
-	if errors.Is(err, gorm.ErrRecordNotFound) {
-		log.Errorf("could not found version endpoint with id %s and error: %v", endpointArg.ID, err)
-		return err
-	}
-	if err != nil {
-		log.Errorf("could not fetch version endpoint with id %s and error: %v", endpointArg.ID, err)
-		// If error getting record from db, return err as RetryableError to enable retry
-		return queue.RetryableError{Message: err.Error()}
-	}
+	endpoint := jobArgs.Endpoint
+	previousStatus := endpoint.Status
```

Comment:
what is the status at the point of request?

Reply:
Depends on whether this is the first deployment or redeployment. If it's first deployment, it would be in Pending state. However, if it's redeployment it would use the current endpoint status.

Reply:
Should we set the status to "pending" regardless of whether it's a deployment or redeployment, while the deployment is in progress? This is the current behavior before this PR, right? (Also relates to the comment below: `// if this is a redeployment, we don't want to update the endpoint status`)

HAving the "pending" status in the UI is helpful to know that a deployment is in progress. Otherwise, the only place is the new history tab which is not very obvious. Wdyt?

Reply:
Oh, my bad. I have updated the endpoint status to be pending regardless of deployment or redeployment: https://github.com/caraml-dev/merlin/pull/479/files/680513818f33a595931250fcb7e1ef41e2dbd684#diff-1a4360ee4934995aec6c53cb8d49a852273a8b6f44cf4cda36a451436ab05b28R120

This is slightly different from the current behaviour where we update all other columns. This PR only updates the endpoint status to pending prior to the deployment/redeployment. Once the deployment is successful, we update other columns. And yes this pending status will also be displayed in the UI.

Reply:
So the [`tobeDeployedEndpoint`](https://github.com/caraml-dev/merlin/pull/479/files/680513818f33a595931250fcb7e1ef41e2dbd684#diff-1a4360ee4934995aec6c53cb8d49a852273a8b6f44cf4cda36a451436ab05b28R133) passed into this Deploy() function has the current endpoint status. While separately we updated endpoint status to pending in DB -- but this will not affect the `tobeDeployedEndpoint`'s status.